### PR TITLE
Log stdout to event log on failure to start process for ANCM out of process

### LINKED ISSUE #10005: [ANCM] [out of process] Write stdout to EventLog when exit code is non-zero
Epic https://github.com/aspnet/AspNetCore/issues/8833

To better diagnose pre-main errors, we should buffer whatever's written to stdout and write to EventLog in event of a non-zero exit code.

### PR DESCRIPTION
**Pull request by jkotalik**
Still thinking about how to simplify some event logs (kind of spammy right now). Will mark out of draft appropriately. 

Fixes https://github.com/aspnet/AspNetCore/issues/10005

### CODE REVIEW

**Comment by jkotalik** [comment]
This be green now 😄

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]:
```suggestion
    while (m_charactersLeft > 0)
```

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]:
```suggestion
        LOG_INFO(L"Canceling standard stream pipe reader.");
```

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: Should something be done here? Otherwise might as well just return without checking last error.

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: nit: should be able to pass the initial size in the string constructor and save the initial small buffer allocation.

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/resources.h
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLib/resources.h]: nit: Add the period back

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: nit: store 4096 in a local variable

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: super nit: so many allocations here, could make a `to_wide_string` function that takes the string and count so you don't need to `substr`

**Review by BrennanConroy — COMMENTED** [review]
Should there be a test that verifies only the first 30kb of output is logged?

**Comment by jkotalik** [comment]
> Should there be a test that verifies only the first 30kb of output is logged?

Yeah we should. Laziness on my part 🐋

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (jkotalik) [reply src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: I think we shouldn't handle errors here. There are at least 3 different errors that could be returned that are false positives. We also cancel reading once the process exits.

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (jkotalik) [reply src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: I wasn't able to find a constructor that took in just a size for string.

**Comment by jkotalik** [comment]
🆙 📅

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp]: Use the new overload!

**Review by BrennanConroy — APPROVED** [review]

**Comment by jkotalik** [comment]
I would like to profile this before merging. I'm concerned that setting these to handles will have a perf impact if we are logging in a middleware.

**Comment by jkotalik** [comment]
Verified locally, no performance regression when logging informational on localhost. I'll still like to do a second pass (and I'll add the accepted label eventually) after we merge this and it is available in the hosting bundle. I'm most concerned about performance on azure, so I'll make sure to verify it there.

I'll merge once this is green.

**Pull request merged**
